ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/services/DefaultKsqlClient.java

```diff
@@ -164,5 +167,50 @@ private static HttpClientOptions createClientOptions() {
     return clientProps;
   }
 
+  private static Consumer<HttpClientOptions> prepareHttpOptionsForSSL(
+      final Map<String, String> clientProps, final boolean verifyHost) {
+    return (httpClientOptions) -> {
+      httpClientOptions.setVerifyHost(verifyHost);
+      httpClientOptions.setSsl(true);
+      final String internalAlias = clientProps
+          .get(KsqlRestConfig.KSQL_SSL_KEYSTORE_ALIAS_INTERNAL_CONFIG);
+      final String trustStoreLocation = clientProps.get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG);
+      if (trustStoreLocation != null) {
+        final String suppliedTruststorePassword = clientProps
+            .get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG);
+        httpClientOptions.setTrustStoreOptions(new JksOptions().setPath(trustStoreLocation)
+            .setPassword(suppliedTruststorePassword == null ? "" : suppliedTruststorePassword));
+
+        final String keyStoreLocation = clientProps.get(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG);
+        if (keyStoreLocation != null) {
```

Comment:
```suggestion
        if (keyStoreLocation != null && !keyStoreLocation.isEmpty()) {
```

??? 

PITA these configs don't default to "" and have a validator to ensure they're not null!